test/integration/transaction.test.js

```diff
@@ -129,6 +129,20 @@ describe(Support.getTestDialectTeaser('Transaction'), function() {
     ).to.eventually.be.rejected;
   });
 
+  it('does not allow queries immediatly after commit call', function() {
+    var self = this;
+    return expect(
+      this.sequelize.transaction().then(function(t) {
+        return self.sequelize.query('SELECT 1+1', {transaction: t, raw: true}).then(function() {
+          return Promise.join(
+            expect(t.commit()).to.eventually.be.fulfilled,
+            expect(self.sequelize.query('SELECT 1+1', {transaction: t, raw: true})).to.eventually.be.rejectedWith(/commit has been called on this transaction\([^)]+\), you can no longer use it/)
+          );
+        });
+      })
+    ).to.be.eventually.fulfilled;
```

Comment:
I had to use this style for assertions because not using the ```expect()``` wrapping inside ```Promise.join(...)``` causes the query to be rejected before the rollback/commit query is finished (since we're using an immediate call after). Thus, this prevents getting the 'There are still running queries' error thrown by ```sequelize.test.verifyNoRunningQueries()``` in ```afterEach()``` callback.
I utilized this approach following the suggestion on http://chaijs.com/plugins/chai-as-promised (see 'Multiple Promise Assertions' section).